### PR TITLE
Refactor/authentication.

### PR DESCRIPTION
**Pull request by emanueljoivo**
Possible breaking changes:

- This backend is now responsible for the OAuth2 authentication and manages the token properly. 
- All endpoints used for manages OAuth2 tokens were removed.
- Only one endpoint to authentication `/auth/oauth2`.

Safe changes: 

- Some documentation updates.
- Dead and legacy code removed.

### CODE REVIEW

File: src/main/java/org/fogbowcloud/app/api/http/services/AuthService.java [hunk 1]
```diff
@@ -92,13 +72,16 @@ public String getValue() {
             });
 
             try {
-                final String oauthTokenRawResponse = HttpWrapper.doRequest(HttpPost.METHOD_NAME, requestUrl,
+                final String oAuthTokenRawResponse = HttpWrapper.doRequest(HttpPost.METHOD_NAME, requestUrl,
                                 headers, null);
-                if (oauthTokenRawResponse != null) {
-                    OAuthToken oAuthToken = gson.fromJson(oauthTokenRawResponse, OAuthToken.class);
+                if (oAuthTokenRawResponse != null) {
+                    OAuthToken oAuthToken = gson.fromJson(oAuthTokenRawResponse, OAuthToken.class);
```
Comment (wesleymonte) [inline src/main/java/org/fogbowcloud/app/api/http/services/AuthService.java]: Could you separate this into a method, for example "requestOAuthToken"?

File: src/main/java/org/fogbowcloud/app/api/http/services/AuthService.java [hunk 2]
```diff
@@ -92,13 +72,16 @@ public String getValue() {
             });
 
             try {
-                final String oauthTokenRawResponse = HttpWrapper.doRequest(HttpPost.METHOD_NAME, requestUrl,
+                final String oAuthTokenRawResponse = HttpWrapper.doRequest(HttpPost.METHOD_NAME, requestUrl,
```
Comment (wesleymonte) [inline src/main/java/org/fogbowcloud/app/api/http/services/AuthService.java]: Could you create a method to generate the authorization header?

**Comment by emanueljoivo** [comment]
Do not merge this right now. We need to provide some way to update the ownCloud's access token automatically.

**Comment by emanueljoivo** [comment]
Branch archived.

**Pull request closed without merging**